accepted/future-releases/nnbd/feature-specification.md

```diff
@@ -310,42 +311,51 @@ and $S$ is \VOID{} or \code{\VOID*}.
 ```
 
 _Comparing to Dart before null-safety, this means that it is no longer allowed
-to "return void to null" in a regular function._
+to return a void expression in a regular function if the return type is
+`Null`._
 
-At [this location](https://github.com/dart-lang/language/blob/65b8267be0ebb9b3f0849e2061e6132021a4827d/specification/dartLangSpec.tex#L15525)
+At [this location](https://github.com/dart-lang/language/blob/65b8267be0ebb9b3f0849e2061e6132021a4827d/specification/dartLangSpec.tex#L15507)
 about an asynchronous non-generator function with future value type `$T_v$`,
 the text is changed as follows:
 
 ```
+It is a compile-time error if $s$ is \code{\RETURN{};},
+unless $T_v$
+is \VOID, \DYNAMIC, or \code{Null}.
+%
+It is a compile-time error if $s$ is \code{\RETURN{} $e$;},
+$T_v$ is \VOID,
+and \flatten{S} is neither \VOID, \DYNAMIC, \code{Null},
+\code{\VOID*}, \code{\DYNAMIC*}, nor \code{Null*}.
+%
 It is a compile-time error if $s$ is \code{\RETURN{} $e$;},
 $T_v$ is neither \VOID{} nor \DYNAMIC,
 and \flatten{S} is \VOID{} or \code{\VOID*}.
-```
-
-_Comparing to Dart before null-safety, this means that it is no longer allowed
-to "return void to null" in an `async` function, nor to "return a void future
-to null"._
-
-The next sentence is changed as follows:
-
-```
+%
```

Comment:
What if `T_v` is `\DYNAMIC*` or `\VOID*`?  See what I mean?  There's no end to it.

Reply:
Indeed. ;-)